drake/solvers/CMakeLists.txt

```diff
@@ -35,8 +35,14 @@ endif()
 
 if(snopt_c_FOUND AND MATLAB_FOUND)
   add_mex(NonlinearProgramSnoptmex NonlinearProgramSnoptmex.cpp)
+  #add_executable(TestSnoptC NPCPP/testSnoptC.cpp)
+  add_executable(TestNonlinearProgram NPCPP/testNonlinearProgram.cpp NPCPP/NonlinearProgram.cpp NPCPP/LinearConstraint.cpp NPCPP/QuadraticConstraint.cpp NPCPP/FunctionHandleConstraint NPCPP/BoundingBoxConstraint.cpp NPCPP/ConstantConstraint.cpp NPCPP/Constraint.cpp)
   pods_use_pkg_config_packages(NonlinearProgramSnoptmex snopt_c)
+  #pods_use_pkg_config_packages(TestSnoptC snopt_c)
+  pods_use_pkg_config_packages(TestNonlinearProgram snopt_c)
   set_target_properties(NonlinearProgramSnoptmex PROPERTIES COMPILE_FLAGS -fPIC)
+  #set_target_properties(TestSnoptC PROPERTIES COMPILE_FLAGS -fPIC)
+  set_target_properties(TestNonlinearProgram PROPERTIES COMPILE_FLAGS -fPIC)
 
   # link against matlab's copy of the gfortran library, if possible
   find_library(GFORTRAN_LIBRARY NAMES gfortran libgfortran
```

Comment:
It looks like you are correct. I removed these links, and the code still built and ran properly. Good thing to know moving forward.